src/sgame/sg_team.cpp

```diff
@@ -287,19 +287,11 @@ void G_ChangeTeam( gentity_t *ent, team_t newTeam )
 	ent->client->pers.classSelection = PCL_NONE;
 	ClientSpawn( ent, nullptr, nullptr, nullptr );
 
-	if ( oldTeam == TEAM_HUMANS && newTeam == TEAM_ALIENS )
+	if ( G_IsPlayableTeam( oldTeam ) && G_IsPlayableTeam( newTeam ) )
 	{
-		// Convert from human to alien credits
 		ent->client->pers.credit =
-		  ( int )( ent->client->pers.credit *
-		           ALIEN_MAX_CREDITS / HUMAN_MAX_CREDITS + 0.5f );
-	}
-	else if ( oldTeam == TEAM_ALIENS && newTeam == TEAM_HUMANS )
-	{
-		// Convert from alien to human credits
-		ent->client->pers.credit =
-		  ( int )( ent->client->pers.credit *
-		           HUMAN_MAX_CREDITS / ALIEN_MAX_CREDITS + 0.5f );
+		  static_cast<int>( ceil( ent->client->pers.credit *
```

Comment:
This calculation is equivalent to `ent->client->pers.credit*g_teamMaxCredits[new_team]/g_teamMaxCredits[oldTeam] + 1`, which I suspect is no what you wanted. (The old calculation didn't make much sense either - it is equivalent to `ent->client->pers.credit*ALIEN_MAX_CREDITS/HUMAN_MAX_CREDITS` and vice versa.)

Reply:
My point was that the math does not make sense - it is doing useless extra operations.

> In my opinion, this whole idea of converting credits is bad, though.

HUMAN_MAX_CREDITS and ALIEN_MAX_CREDITS are the same, so you could certainly consider the conversion idea.

Reply:
you meant, consider the removal of conversion idea? Looks like I didn't noticed at that point that the constants were the same... dunno why, since I think I merged them into something too in some commit? maybe one not candidate for push? Don't remember, but this really looks like code which could be avoided.